File: src/main/java/com/example/timetable/config/security/CustomUserDetailsService.java
package com.example.timetable.config.security;

import com.example.timetable.profile.ProfileEntity;
import com.example.timetable.profile.ProfileRepository;
import lombok.AllArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
@AllArgsConstructor
public class CustomUserDetailsService implements UserDetailsService {

    private final ProfileRepository profileRepository;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        // adminjon_mazgi@gmail.com
        System.out.println("loadUserByUsername");
        Optional<ProfileEntity> optional = profileRepository.findByEmail(username);
        if (optional.isEmpty()) {
            throw new UsernameNotFoundException("User not found");
        }
        ProfileEntity profile = optional.get();
        return new CustomUserDetails(profile);
    }
}

File: src/main/kotlin/com/example/w3schooluz/config/CustomUserDetailsService.kt
package com.example.w3schooluz.config

import com.example.w3schooluz.profile.ProfileEntity
import com.example.w3schooluz.profile.ProfileRepository
import lombok.AllArgsConstructor
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service
import java.util.*


@Service
@AllArgsConstructor
class CustomUserDetailsService(val profileRepository: ProfileRepository) : UserDetailsService {
    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): UserDetails {
        // adminjon_mazgi@gmail.com
        val optional: Optional<ProfileEntity> = profileRepository.findByEmail(username)
        if (optional.isEmpty) {
            throw UsernameNotFoundException("User not found")
        }
        val profile: ProfileEntity = optional.get()
        return CustomUserDetails(profile)
    }
}